src/main/java/io/pinecone/clients/Index.java

```diff
@@ -98,19 +98,63 @@ public QueryResponseWithUnsignedIndices queryByVectorId(int topK,
         return query(topK, null, null, null, id, namespace, filter, false, false);
     }
 
+    @Override
+    public QueryResponseWithUnsignedIndices queryByVectorId(int topK,
+                                                            String id,
+                                                            String namespace,
+                                                            boolean includeValues,
+                                                            boolean includeMetadata) {
+        return query(topK, null, null, null, id, namespace, null, includeValues, includeMetadata);
+    }
+
     @Override
     public QueryResponseWithUnsignedIndices queryByVectorId(int topK,
                                                             String id,
                                                             String namespace) {
         return query(topK, null, null, null, id, namespace, null, false, false);
     }
 
+    @Override
+    public QueryResponseWithUnsignedIndices queryByVectorId(int topK, String id, boolean includeValues, boolean includeMetadata) {
+        return query(topK, null, null, null, id, null, null, includeValues, includeMetadata);
+    }
+
     @Override
     public QueryResponseWithUnsignedIndices queryByVectorId(int topK,
                                                             String id) {
         return query(topK, null, null, null, id, null, null, false, false);
     }
 
+    @Override
+    public QueryResponseWithUnsignedIndices queryByVector(int topK, List<Float> vector, String namespace, Struct filter, boolean includeValues, boolean includeMetadata) {
```

Comment:
nit: if we can have each parameter on new line just like other methods have it and the asyncIndex does for the same method, it'll be easy to read but totally optional